crates/compiler/fmt/src/expr.rs

```diff
@@ -92,13 +92,17 @@ fn format_expr_only(
             buf.indent(indent);
             buf.push_str("try");
         }
+        Expr::Apply(loc_expr, loc_args, called_via::CalledVia::ParensAndCommas) => {
+            fmt_apply(loc_expr, loc_args, indent, buf, true);
+        }
         Expr::Apply(loc_expr, loc_args, _) => {
             let apply_needs_parens = parens == Parens::InApply || parens == Parens::InApplyLastArg;
-
-            if apply_needs_parens && !loc_args.is_empty() {
+            if buf.flags().parens_and_commas {
+                fmt_apply(loc_expr, loc_args, indent, buf, false);
+            } else if apply_needs_parens && !loc_args.is_empty() {
                 fmt_parens(item, buf, indent);
             } else {
-                fmt_apply(loc_expr, loc_args, indent, buf);
+                fmt_apply(loc_expr, loc_args, indent, buf, false);
```

Comment:
The first and last branches here could be combined - e.g. `if buf.flags().parens_and_commas || !(apply_needs_parens && !loc_args.is_empty())`